module-2/cdk/lib/ECSConstruct.ts

```diff
@@ -0,0 +1,179 @@
+import ec2 = require("@aws-cdk/aws-ec2");
+import ecs = require("@aws-cdk/aws-ecs");
+import ecsPatterns = require("@aws-cdk/aws-ecs-patterns");
+import iam = require("@aws-cdk/aws-iam");
+import { Role } from "@aws-cdk/aws-iam";
+import cdk = require("@aws-cdk/cdk");
+import { CoreConstruct } from "./CoreConstruct";
+import { EcrConstruct } from "./EcrConstruct";
+
+export class EcsClusterConstruct extends cdk.Construct {
+  public readonly cluster: ecs.Cluster;
+  public readonly ecsService: LoadBalancedFargateService;
+
+  constructor(scope: cdk.Construct, id: string, core: CoreConstruct, ecr: EcrConstruct) {
+    super(scope, id);
+
+    // This is a role which is used by the ECS tasks. Tasks in Amazon ECS define
+    // the containers that should be deployed togehter and the resources they
+    // require from a compute/memory perspective. So, the policies below will define
+    // the IAM permissions that our Mythical Mysfits docker containers will have.
+    // If you attempted to write any code for the Mythical Mysfits service that
+    // interacted with different AWS service APIs, these roles would need to include
+    // those as allowed actions.
+    const ECSTaskRole = new iam.Role(this, "ECSTaskRole", {
+      assumedBy: new iam.ServicePrincipal("ecs-tasks.amazonaws.com"),
+      path: "/"
+    });
+
+    const ECSTaskRoleStatement1 = new iam.PolicyStatement();
```

Comment:
Use "grant" methods instead of these. Let us know if there is missing coverage and we'll add it. In general, we don't want to see any explicit low-level IAM statements...